appcelerator/titanium_mobile
# feat: add "isTrusted" property to Slider "change" event

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27977

**Summary:**
Added a "isTrusted" boolean property to `Ti.UI.Slider` object's "change" event. Property will be `true` if change came from the end-user. Will be `false` if change was made programmatically. This is a standard JavaScript event property available to web developers. (In the future, we should apply this property to all events.)
https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted

**Test:**
1. Build and run the below for Android.
2. Drag the slider with your finger.
3. In the log, verify change event's `isTrusted` property is set to `true`.
4. Tap on the "Update Slider" button.
5. In the log, verify change event's `isTrusted` property is set to `false`.

```javascript
var window = Ti.UI.createWindow();
var slider = Ti.UI.createSlider({
	min: 0.0,
	max: 1.0,
	value: 0.25,
});
slider.addEventListener("change", function(e) {
	Ti.API.info("@@@ slider value changed: " + e.value + ", isTrusted: " + e.isTrusted);
});
window.add(slider);
var button = Ti.UI.createButton({
	title: "Update Slider Position",
	bottom: "20dp",
});
button.addEventListener("click", function() {
	slider.value = 0.5;
});
window.add(button);
window.open();
```
